### PR TITLE
use route_settings for hidden and operations extensions

### PR DESCRIPTION
**Pull request by thogg4**
@LeFnord This pr may be a little heavy handed, but I wanted to show you what I meant by having grape-swagger use the preferred way to pass route settings.

Please let me know what you think and what I can improve and change.

Addresses https://github.com/ruby-grape/grape-swagger/issues/594

### CODE REVIEW

**Comment by LeFnord** [comment]
will give it a try and a deeper look on the weekend, ok?

**Comment by thogg4** [comment]
@LeFnord what do you think about this approach?

File: grape-swagger.gemspec [hunk 1]
```diff
@@ -14,7 +14,7 @@ Gem::Specification.new do |s|
   s.license     = 'MIT'
 
   s.required_ruby_version = '>= 2.2.6'
-  s.add_runtime_dependency 'grape', '>= 0.16.2'
+  s.add_runtime_dependency 'grape', '>= 0.19.1'
```
Comment (LeFnord) [inline grape-swagger.gemspec]: please revert it back

File: lib/grape-swagger/endpoint.rb [hunk 2]
```diff
@@ -337,7 +337,8 @@ def model_name(name)
     end
 
     def hidden?(route, options)
-      route_hidden = route.options[:hidden]
+      route_hidden = route.settings.try(:[], :swagger).try(:[], :hidden)
+      route_hidden = route.options[:hidden] if route.options.key?(:hidden)
```
Comment (LeFnord) [inline lib/grape-swagger/endpoint.rb]: am I right, the original behavior still available?

File: spec/swagger_v2/api_swagger_v2_extensions_spec.rb [hunk 3]
```diff
@@ -19,10 +19,11 @@ class ExtensionsApi < Grape::API
           { 'declared_params' => declared(params) }
         end
 
+        route_setting :x_operation, some: 'stuff'
```
Comment (LeFnord) [inline spec/swagger_v2/api_swagger_v2_extensions_spec.rb]: I like it, makes the extensions more consistent, thanks

File: lib/grape-swagger/endpoint.rb (hunk 2 above)
Comment (thogg4) [inline lib/grape-swagger/endpoint.rb]: Yeah, the original behavior is default.

**Comment by LeFnord** [comment]
👍 @thogg4 please add the changelog entry, then I'll mörge it

**Pull request merged**